### PR TITLE
Ekirjasto 131 selected books

### PR DESCRIPTION
**Pull request by natlibfi-kaisa**
## Description

Enables a Patron to add and remove books from their list of selected books.

## Motivation and Context

This feature was requested for patrons to save books from the collection.

A short summary of the changes:

- A new table `selected_books` which is also a class `SelectedBook` maps `Patron` and `Work` as a many-to-many relationship. If a `Patron` is removed, their selected books are also removed.

- Two new routes for select and unselect requests are added and a new controller that creates responses to them:
   - `/works/<identifier_type>/<path:identifier>/select_book`: A POST request: When a book is selected, a selected-tag with it's creation date is added to the work's OPDS feed: `<selected>2024-12-16T07:29:16+00:00</selected>`.
   - `/works/<identifier_type>/<path:identifier>/unselect_book`: A DELETE request: When the patron removes the book from their selected books, the tag is removed from the feed.
   - `/selected_books/<identifier_type>/<path:identifier>`: GET and DELETE requests: A selected book's details can be fetched and the book can be deleted from the list.
- A route `/selected_books`: GET request: Returns an OPDS acquisition feed with all the patron's selected books. This feed also contains loan/hold information atm.
- The authentication document includes the link for selected books so it can be easily accessed via the apps.
- Basically, since loans and hold and now selected books are patron dependent annotations, these are always checked and added to the patron's feed.

## How Has This Been Tested?

Locally running the application.

## Checklist

- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [ ] Transifex translators have been notified. --> N/A

### CODE REVIEW

File: api/controller/select_books.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+        feed = OPDSAcquisitionFeed.selected_books_for(self.circulation, patron)
+
+        response = feed.as_response(
+            max_age=0,
+            private=True,
+            mime_types=flask.request.accept_mimetypes,
+        )
+
+        # For loans, the patron's last loan activity sync time was set. Not yet
```
Comment (natlibfi-kaisa) [inline api/controller/select_books.py:31]: Do we see any need for this in the future for selected books? No, I think?

File: core/feed/acquisition.py [hunk 2]
```diff
@@ -511,16 +511,84 @@ def active_loans_for(
...
+        url = annotator.url_for(
+            "selected_books",
+            library_short_name=patron.library.short_name,
+            _external=True,
+        )
+        works = patron.get_selected_works()
+
+        feed = OPDSAcquisitionFeed("Selected books", url, works, annotator)
+        feed.generate_feed()
+        return feed
```
Comment (natlibfi-kaisa) [inline core/feed/acquisition.py:553]: Also contains loan/hold info. Any reason we wouldn't want that here? Doesn't hurt, I guess.

File: api/controller/select_books.py (hunk 1 above)
Comment (NatLibFi-JoonaKupe) [reply api/controller/select_books.py:31]: Can think of any use atm

File: core/feed/acquisition.py (hunk 2 above)
Comment (NatLibFi-JoonaKupe) [reply core/feed/acquisition.py:553]: Could come in handy one day

**Review by NatLibFi-JoonaKupe — APPROVED** [review]

**Review by NatLibFi-JoonaKupe — APPROVED** [review]

**Review by NatLibFi-JoonaKupe — APPROVED** [review]

**Review by natlibfi-burger — APPROVED** [review]
